db/seeds.rb

```diff
@@ -1,8 +1,30 @@
 # frozen_string_literal: true
+
 # This file should contain all the record creation needed to seed the database with its default values.
 # The data can then be loaded with the bin/rails db:seed command (or created alongside the database with db:setup).
 #
 # Examples:
 #
 #   movies = Movie.create([{ name: "Star Wars" }, { name: "Lord of the Rings" }])
 #   Character.create(name: "Luke", movie: movies.first)
+require 'csv'
+
+ProduceItem.destroy_all
+ItemSeason.destroy_all
+
+id_arr = []
+
+CSV.foreach(Rails.root.join('lib/produce.csv'), headers: true) do |row|
+  produce_item = ProduceItem.create({
+    name: row[0],
+    category: row[1]
+  })
+
+  id_arr << produce_item.id
+
+  ItemSeason.create({
+    country_code: row[2],
+    month_index: row[3],
+    produce_item_id: id_arr.shift
```

Comment:
Why are we adding the ID to an array and then removing it to add it here? I tend to think that using the id directly would be easier (i.e `produce_item.id`). WDYT?

On a sidenote, we could probably simplify this by creating the item directly there:

```rb
ItemSeason.create(
  country_code: row[2],
  month_index: row[3],
  produce_item: ProduceItem.create(name: row[0], category: row[1])
)
```

Reply:
I like your solution waaay more. I was trying to figure out how to get the ProduceItem id into ItemSeason and came up with that, I thought it looked a bit wonky but couldn't figure out how to refactor it.

Reply:
I'm trying to run it, but I'm getting an error, not sure if it's related:
`rails aborted!
ActiveRecord::NotNullViolation: PG::NotNullViolation: ERROR:  null value in column "produce_item_id" of relation "item_seasons" violates not-null constraint
DETAIL:  Failing row contains (34, 2, BR, null, 2022-06-09 19:42:24.530141, 2022-06-09 19:42:24.530141).
/Users/manu/code/produce_in_season/db/seeds.rb:18:in `block in <main>'
/Users/manu/code/produce_in_season/db/seeds.rb:17:in `<main>'

Caused by:
PG::NotNullViolation: ERROR:  null value in column "produce_item_id" of relation "item_seasons" violates not-null constraint
DETAIL:  Failing row contains (34, 2, BR, null, 2022-06-09 19:42:24.530141, 2022-06-09 19:42:24.530141).
/Users/manu/code/produce_in_season/db/seeds.rb:18:in `block in <main>'
/Users/manu/code/produce_in_season/db/seeds.rb:17:in `<main>'
Tasks: TOP => db:seed
(See full trace by running task with --trace)`

Reply:
We could create the item before and then assign it.

```rb
item = ProduceItem.create!(name: row[0], category: row[1])
ItemSeason.create!(
  country_code: row[2],
  month_index: row[3],
  produce_item: item
)
```

Added `!` so the program crashes in case there is a validation error.